### PR TITLE
7 add support for custom tunnels

### PR DESCRIPTION
**Pull request by Jack-R-Long**
Steps to test
1. Create tunnel and DNS rules [Docs](https://developers.cloudflare.com/cloudflare-one/connections/connect-networks/get-started/create-local-tunnel/)
2. Supply values to Helm chart
3. Deploy!

### CODE REVIEW

File: templates/cloudflared/_helpers.tpl [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{{/*
+Expand the name of the chart.
+*/}}
+{{- define "cloudflare-tunnel.name" -}}
+{{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" }}
+{{- end }}
```
Comment (willswire) [inline templates/cloudflared/_helpers.tpl:6]: This looks like a nice-to-have, but I don't see if being referenced elsewhere. Maybe we should remove it for now.

File: templates/cloudflared/_helpers.tpl [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+{{/*
+Create a default fully qualified app name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+If release name contains chart name it will be used as a full name.
+*/}}
```
Comment (willswire) [inline templates/cloudflared/_helpers.tpl:12]: Nice catch!

File: values.yaml [hunk 3]
```diff
@@ -21,3 +21,18 @@ cloudflared:
   image:
     repository: "cloudflare/cloudflared"
     tag: "2024.12.2"
+  userTunnel:
+    enabled: false
```
Comment (willswire) [inline values.yaml:25]: Maybe we should drop a comment in here about Quick Tunnels as default, with a link to the Cloudflare docs.

**Review by willswire — CHANGES_REQUESTED** [review]
Overall looks great! Just a quick change on the `_helpers.tpl` file to reduce unnecessary complexity.

File: templates/cloudflared/_helpers.tpl (hunk 1 above)
Comment (Jack-R-Long) [reply templates/cloudflared/_helpers.tpl]: Done!

File: values.yaml (hunk 3 above)
Comment (Jack-R-Long) [reply values.yaml]: Done

**Review by willswire — APPROVED** [review]

**Pull request merged**